### PR TITLE
single-post component

### PR DESCRIPTION
**Pull request by meinou**

### CODE REVIEW

File: apps/blog/src/app/feed/feed.component.html [hunk 1]
```diff
@@ -3,19 +3,20 @@
 
   <mat-card  *ngFor="let post of posts$ | async">
     
-    <mat-card-content >
+    <codelab-single-post [post]=post [full]='false' [ngClass]="text"></codelab-single-post>
+    <!-- <mat-card-content >
       <p class="head">
```
Comment (NothingEverHappens) [inline apps/blog/src/app/feed/feed.component.html]: drop the comment?

File: apps/blog/src/app/single-post/single-post.component.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<!-- 
+<ng-container *ngIf="post$ | async as post">
```
Comment (NothingEverHappens) [inline apps/blog/src/app/single-post/single-post.component.html]: Drop the comment?

File: apps/blog/src/app/single-post/single-post.component.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+
+@Component({
+  selector: 'codelab-single-post',
+  templateUrl: './single-post.component.html',
+  styleUrls: ['./single-post.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SinglePostComponent implements OnInit {
+//  post$: Observable<Post>;
```
Comment (NothingEverHappens) [inline apps/blog/src/app/single-post/single-post.component.ts]: Drop the comment

File: apps/blog/src/app/single-post/single-post.component.html [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<!--
```
Comment (barlanj) [inline apps/blog/src/app/single-post/single-post.component.html]: do we still need these comments?

File: apps/blog/src/app/feed/feed.component.html [hunk 5]
```diff
@@ -3,19 +3,7 @@
...
-      </p>
-      
-      <div class="container">
-        <markdown ngPreserveWhitespaces>
-          <div class="text gradient" [innerHTML]="post.text"></div>
-        </markdown> 
-        <a routerLink="/post/{{ post.key}}">More</a>
-      </div>
-    </mat-card-content>
+    <codelab-single-post [post]=post [full]='false' [ngClass]="text"></codelab-single-post>
```
Comment (NothingEverHappens) [inline apps/blog/src/app/feed/feed.component.html]: double quotes
here and everywhere

File: apps/blog/src/app/single-post/single-post.component.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  selector: 'codelab-single-post',
+  templateUrl: './single-post.component.html',
+  styleUrls: ['./single-post.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SinglePostComponent implements OnInit {
+
+ @Input() post: Post;
+ @Input() full: boolean;
+ @Input() ngClass: any;
```
Comment (NothingEverHappens) [inline apps/blog/src/app/single-post/single-post.component.ts]: ngClass -> className? It's a string. 
Why fo you need this?

File: apps/blog/src/app/single-post/single-post.component.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+ ngOnInit() {}
```
Comment (NothingEverHappens) [inline apps/blog/src/app/single-post/single-post.component.ts]: drop ngOnInit, it's not used?

File: apps/blog/src/app/single-post/single-post.component.html [hunk 8]
```diff
@@ -0,0 +1,17 @@
+  <mat-card-content >
```
Comment (NothingEverHappens) [inline apps/blog/src/app/single-post/single-post.component.html]: Move mat-card inside of the comeponent

File: apps/blog/src/app/single-post/single-post.component.scss [hunk 9]
```diff
@@ -0,0 +1,33 @@
+
```
Comment (NothingEverHappens) [inline apps/blog/src/app/single-post/single-post.component.scss]: delete those empty lines?

File: apps/blog/src/app/single-post/single-post.component.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
+import { ChangeDetectionStrategy, Component, OnInit, Input } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Post } from '../form/form.component';
+import { FormService } from '../form.service';
+import { ActivatedRoute } from '@angular/router';
+
+
```
Comment (NothingEverHappens) [inline apps/blog/src/app/single-post/single-post.component.ts]: Delete empty lines

File: apps/blog/src/app/single-post/single-post.component.ts (hunk 6 above)
Comment (meinou) [reply apps/blog/src/app/single-post/single-post.component.ts]: To change className for different use cases of this component?
Is there a better way?

File: apps/blog/src/app/single-post/single-post.component.ts [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+@Component({
+  selector: 'codelab-single-post',
+  templateUrl: './single-post.component.html',
+  styleUrls: ['./single-post.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SinglePostComponent{
+ @Input() post: Post;
+ @Input() full: boolean;
+ @Input() ngClass: string;
```
Comment (NothingEverHappens) [inline apps/blog/src/app/single-post/single-post.component.ts]: Drop this input and just use a constant

File: apps/blog/src/app/single-post/single-post.component.scss [hunk 12]
```diff
@@ -0,0 +1,31 @@
+.date {
+    float: right;
+    margin-right: 5px;
+  }
```
Comment (NothingEverHappens) [inline apps/blog/src/app/single-post/single-post.component.scss]: Reformat this file

**Pull request merged**